simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,382 @@
+        type = checkNonNegativeFloat,
+        help = 'Variance for likeability of a local thing based on place lived.')
+
+    parser.add_argument(
+        '--outputDir',
+        default = '.',
+        type = str,
+        help = 'Output directory to which all data files will be written.')
+
+    parser.add_argument(
+        '--people',
+        default = DEFAULT_PEOPLE,
+        type = checkNonNegativeInt,
+        help = 'Number of people in your dataset.')
+
+    parser.add_argument(
+        '--places',
+        default = DEFAULT_PLACES,
+        type = checkNonNegativeInt,
+        help = 'Number of places in your dataset.')
+
+    parser.add_argument(
+        '--placesLivedMean',
+        default = DEFAULT_PLACES_LIVED_MEAN,
+        type = checkNonNegativeInt,
+        help = 'Mean of the number of places a person has lived in.')
+
+    parser.add_argument(
+        '--placesLivedSD',
+        default = DEFAULT_PLACES_LIVED_SD,
+        type = checkNonNegativeFloat,
+        help = 'Standard deviation of the number of places a person has lived in.')
+
+    parser.add_argument(
+        '--seed',
+        default = numpy.random.randint(DEFAULT_MAX_RAND_INT),
+        type = int,
+        help = 'Seed for the random number generator used in the script.')
+
+    parser.add_argument(
+        '--targetSplit',
+        default = DEFAULT_TARGET_SPLIT,
+        type = checkValidSplit,
+        help = 'Fraction of the truth data to be used as targets.')
+
+    return parser.parse_args()
+
+def getTruncatedNormal(mean, sd, lower, upper):
+    """
+    Return truncated normal distribution.
+    """
+    return scipy.stats.truncnorm(
+        (lower - mean) / sd, (upper - mean) / sd, loc = mean, scale = sd)
+
+class Person():
+    """
+    Person class for the dataset.
+    """
+
+    def __init__(self):
+        self.index = None
+        self.globalLikes = None
+        self.lived = None
+        self.localLikes = None
+
+class DataGen():
+    """
+    Data generator class for simple-acquaintances example.
+    """
+
+    def __init__(self, numberOfPeople, numberOfPlaces, numberOfGlobal, numberOfLocal,
+            placesLivedMean, placesLivedSD, localLikesVariance, targetSplit,
+            seed, outputDir):
+        self.globalThings = [i for i in range(numberOfGlobal)]
+        self.localLikesVariance = localLikesVariance
+        self.localThings = [i for i in range(numberOfLocal)]
+        self.numberOfPeople = numberOfPeople
+        self.outputDir = outputDir
+        self.places = [i for i in range(numberOfPlaces)]
+        self.placesLivedMean = placesLivedMean
+        self.placesLivedSD = placesLivedSD
+        self.placesLivedUpper = numberOfPlaces
+        self.seed = seed
+        self.rng = self.__createRNG(seed)
+        self.targetSplit = targetSplit
+
+        self.placesAffectingLocalThings = {}
+        self.__generatePALT()  # Fill in placesAffectingLocalThings.
+
+
+    def __createRNG(self, seed):
```

Comment:
Use only a single underscore for private methods.
Double underscores are for python builtins.